main.go

```diff
@@ -55,12 +67,63 @@ func main() {
 		}()
 	}
 
-	go func() {
-		// no shutdown mechanism in place for the HTTP server
-		componentDone <- http.Start(server, *listenAddress)
-	}()
+	for _, tcpCheckerAddress := range *tcpCheckerAddresses {
+		values := strings.Split(tcpCheckerAddress, ":")
+		var name, address, portStr string
+
+		if len(values) == 3 {
+			name = values[0]
+			address = values[1]
+			portStr = values[2]
+		} else if len(values) == 2 {
+			address = values[0]
+			portStr = values[1]
+			name = fmt.Sprintf("%s:%s", address, portStr)
+		} else {
+			log.Errorf("Could not understand tcp-checker %s", tcpCheckerAddress)
+			os.Exit(1)
+		}
+		port, err := strconv.ParseInt(portStr, 10, 32)
+		if err != nil {
+			log.Errorf("Could not parse port %s as integer in tcp-checker %s. Error: %s", portStr, tcpCheckerAddress, err)
+			os.Exit(1)
+		}
+
+		logger := log.
+			With("type", "tcpchecker").
+			With("name", name).
+			With("address", address).
+			With("port", port)
+		logger.Infof("Start checking address %s:%v", address, port)
+		tcpCheckerDaemon := daemon.New(daemon.Configuration{
+			Logger:   logger,
+			Interval: 1 * time.Second,
+			Tick: func() {
+				tcpchecker.Check(name, address, int(port), tcpchecker.CompositeReporter{
+					Reporters: []tcpchecker.Reporter{
+						&tcpchecker.LogReporter{
+							Logger: logger,
+						},
+						reporter.TcpChecker,
+					},
```

Comment:
Isn't `metrics.PrometheusReporter` missing here?

Reply:
That's the `reporter.TcpChecker`, right? Maybe the variable should be `prometheusReporter` instead.

Reply:
Ah, might be. Renaming would make sense 👍